File: frontend/src/services/InventoryService.js
import API from "./api";

export const getInventory = () => {
    return API.get("/api/inventory");
};

export const getInventoryById = (id) => {
    return API.get(`/api/inventory/${id}`);
};

export const addInventory = (inventory) => {
    return API.post("/api/inventory", inventory);
};

export const updateInventory = (id, inventory) => {
    return API.put(`/api/inventory/${id}`, inventory);
};

export const deleteInventory = (id) => {
    return API.delete(`/api/inventory/${id}`);
};

File: src/main/java/com/example/lab_resource_platform/service/InventoryService.java
package com.example.lab_resource_platform.service;

import com.example.lab_resource_platform.entity.Inventory;
import java.util.List;

public interface InventoryService {

    Inventory addInventory(Inventory inventory);

    List<Inventory> getAllInventory();

    Inventory getInventoryById(Long id);

    Inventory updateInventory(Long id, Inventory inventory);

    void deleteInventory(Long id);

}